phpBB/phpbb/report/controller/report.php

```diff
@@ -0,0 +1,268 @@
+	 */
+	protected $request;
+
+	/**
+	 * @var \phpbb\captcha\factory
+	 */
+	protected $captcha_factory;
+
+	/**
+	 * @var string
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * @var string
+	 */
+	protected $php_ext;
+
+	/**
+	 * @var array
+	 */
+	protected $tmp_vars;
+
+	/**
+	 * @var \phpbb\report\report_handler_interface
+	 */
+	protected $report_handler;
+
+	/**
+	 * @var \phpbb\report\report_ui_provider
+	 */
+	protected $report_ui_provider;
+
+	public function __construct(\phpbb\config\db $config, \phpbb\user $user, \phpbb\template\template $template, \phpbb\controller\helper $helper, \phpbb\request\request_interface $request, \phpbb\captcha\factory $captcha_factory, \phpbb\report\factory $report_factory, \phpbb\report\report_ui_provider $ui_provider, $phpbb_root_path, $php_ext)
+	{
+		$this->config			= $config;
+		$this->user				= $user;
+		$this->template			= $template;
+		$this->helper			= $helper;
+		$this->request			= $request;
+		$this->phpbb_root_path	= $phpbb_root_path;
+		$this->php_ext			= $php_ext;
+		$this->captcha_factory	= $captcha_factory;
+		$this->tmp_vars			= array();
+		$this->report_handler	= $report_factory;
+
+		$this->report_ui_provider = $ui_provider;
+	}
+
+	/**
+	 * Controller for /report/{module}/{id} routes
+	 *
+	 * @param string	$mode
+	 * @param int		$id
+	 * @return \Symfony\Component\HttpFoundation\Response a Symfony response object
+	 * @throws \phpbb\exception\http_exception when $mode or $id is invalid for some reason
+	 */
+	public function handle($mode, $id)
+	{
+		// Get report handler
+		$this->report_handler = $this->report_handler->get($mode);
+
+		$this->user->add_lang('mcp');
+
+		// Set up variables
+		$this->tmp_vars = array(
+			'mode'		=> $mode,
+			'id'		=> $id,
+			'form'		=> array(
+				'error'				=> array(),
+				's_hidden_fields'	=> '',
+				'report_text'		=> $this->request->variable('report_text', '', true),
+				'reason_id'			=> $this->request->variable('reason_id', 0),
+				'user_notify'		=> ($this->user->data['is_registered']) ? $this->request->variable('notify', 0) : false,
+				'submit'			=> $this->request->variable('submit', ''),
+				'cancel'			=> $this->request->variable('cancel', ''),
+			),
+		);
+
+		$redirect_url = append_sid(
+			$this->phpbb_root_path . ( ($mode === 'pm') ? 'ucp' : 'viewtopic' ) . ".{$this->php_ext}",
+			($mode == 'pm') ? "i=pm&mode=view&p=$id" : "p=$id"
+		);
+		$redirect_url .= ($mode === 'post') ? "#p$id" : '';
+
+		// Set up CAPTCHA
+		if ($this->config['enable_post_confirm'] && !$this->user->data['is_registered'])
+		{
+			$this->tmp_vars['captcha'] = $this->captcha_factory->get_instance($this->config['captcha_plugin']);
+			$this->tmp_vars['captcha']->init(CONFIRM_REPORT);
+		}
+
+		//Has the report been cancelled?
+		if (!empty($this->tmp_vars['form']['cancel']))
+		{
+			redirect($redirect_url);
+		}
+
+		// Check CAPTCHA
+		if (!empty($this->tmp_vars['form']['submit']))
+		{
+			if (isset($this->tmp_vars['captcha']))
```

Comment:
extract the whole content of the if (so the condition and the next one) to a dedicated method